src/coreclr/debug/createdump/threadinfo.cpp

```diff
@@ -53,6 +53,16 @@ ThreadInfo::UnwindNativeFrames(CONTEXT* pContext)
         uint64_t ip = 0, sp = 0;
         GetFrameLocation(pContext, &ip, &sp);
 
+#if defined(__aarch64__)
+        // ARM64 can have frames with the same SP but different IPs. Increment sp so it gets added to the stack
+        // frames in the correct order and to prevent the below loop termination on non-increasing sp. Since stack
+        // pointers are always 8 byte align, this increase is masked off in StackFrame::StackPointer() to get the
+        // original stack pointer.
+        if (sp == previousSp && ip != previousIp)
```

Comment:
I'm just curious - where do we order the stack frames? Cannot we just keep the original order we've extracted them instead of introducing this mechanism? I am a bit worried about the fact that if there were more than 8 such frames in a consecutive order, we would not be able to handle it correctly.

Reply:
This is some randomness in the way createdump enum's native frames.  It enums all the native frames before enums managed frames and between each managed frame.  So the thread set of frames is ordered by SP which also keeps the dups from being added. 

Later in 7.0, I want to revisit this thread stack frame set and the frame enumeration to see if it can just be a list and only enum native frames before/after the managed frame enum and not between each managed frame.

Reply:
We should never get more than 2 frames in a row with the same SP I think because LR would need to be restore from the stack (thus inc'ing SP) on the 3rd frame (unless it can be restored from another register).

Reply:
It sounds you are right.